micrometer-core/src/main/java/io/micrometer/core/instrument/binder/tomcat/ObservedValve.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.tomcat;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+
+import org.apache.catalina.Valve;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
+import org.apache.catalina.valves.ValveBase;
+import io.micrometer.core.instrument.binder.http.DefaultHttpServerRequestObservationConvention;
+import io.micrometer.core.instrument.binder.http.HttpObservationDocumentation;
+import io.micrometer.core.instrument.binder.http.HttpServerRequestObservationContext;
+import io.micrometer.core.instrument.binder.http.HttpServerRequestObservationConvention;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+
+/**
+ * A {@link Valve} that creates {@link Observation}.
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.12.0
+ * @see HttpObservationDocumentation
+ */
+public class ObservedValve extends ValveBase {
```

Comment:
Some guidance in the JavaDoc on when a user would configure this instrumentation would be good. Users should avoid double instrumentation that can happen when configuring instrumentation at multiple layers in the same stack for HTTP request processing. For example, I would expect users should not configure this and servlet-level or jax-rs or Spring WebMvc instrumentation.